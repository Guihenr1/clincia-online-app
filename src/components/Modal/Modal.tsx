import { FC, ReactNode, MouseEvent } from "react";
import useStyles from "./styles";
import { Box, Modal as ModalMui, Typography } from "@mui/material";
import CloseIcon from "@mui/icons-material/Close";
import Button from "../Button/Button";

interface ModalProps {
  children: ReactNode;
  open: boolean;
  handleClose: () => void;
  title?: string;
}

const Modal: FC<ModalProps> = ({ children, open, handleClose, title }) => {
  const s = useStyles();

  return (
    <div>
      <ModalMui open={open} onClose={handleClose} data-testid="modal">
        <Box className={s.classes.children}>
          <Box className={s.classes.header}>
            <Typography variant="h5">{title}</Typography>
            <CloseIcon
              onClick={handleClose}
              className={s.classes.closeButton}
            />
          </Box>
          <Box className={s.classes.content}>{children}</Box>
          <Box className={s.classes.bottom}>
            <Button className={s.classes.cancel} color="warning">
              Cancel
            </Button>
            <Button>Confirm</Button>
          </Box>
        </Box>
      </ModalMui>
    </div>
  );
};

Modal.defaultProps = {
  children: "",
  title: "",
};

export default Modal;
