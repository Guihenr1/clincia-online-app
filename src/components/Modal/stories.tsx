import { Meta } from "@storybook/react";
import Modal from "./Modal";
import { useState } from "react";
import Button from "../Button/Button";

export default {
  component: Modal,
  args: {
    children: <p>TEST</p>,
  },
} as Meta;

export const Template = (args: any) => {
  const [open, setOpen] = useState(true);

  return (
    <>
      <Button onClick={() => setOpen(true)}>OPEN</Button>
      <Modal
        {...args}
        title="Test"
        open={open}
        handleClose={() => setOpen(false)}
      />
    </>
  );
};
