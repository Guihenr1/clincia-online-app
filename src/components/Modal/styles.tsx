import { makeStyles } from "tss-react/mui";

const useStyles = makeStyles()(() => ({
  children: {
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    minWidth: "300px",
    boxShadow: "24",
    borderRadius: "10px",
    backgroundColor: "#fff",
  },
  header: {
    height: "50px",
    borderBottom: "solid 1px #ccc",
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
    padding: "10px",
  },
  bottom: {
    height: "50px",
    borderTop: "solid 1px #ccc",
    display: "flex",
    alignItems: "center",
    marginLeft: "auto",
    padding: "10px",
    justifyContent: "end",
  },
  closeButton: {
    cursor: "pointer",
  },
  content: {
    padding: "10px",
  },
  cancel: {
    marginRight: "10px",
  },
}));

export default useStyles;
