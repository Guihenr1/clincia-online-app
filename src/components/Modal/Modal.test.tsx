import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import Modal from "./Modal";
import "@testing-library/jest-dom/extend-expect";

describe("Modal", () => {
  let open = true;
  const onClickClose = jest.fn();
  const children = "test";

  test("renders the Modal", () => {
    const { container } = render(
      <Modal open={open} handleClose={onClickClose}>
        {children}
      </Modal>
    );

    expect(screen.getByTestId("modal")).toBeInTheDocument();
    expect(screen.getByText(children)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("not renders the Modal", async () => {
    render(
      <Modal open={false} handleClose={onClickClose}>
        {children}
      </Modal>
    );

    await waitFor(() => {
      expect(screen.queryByTestId("modal")).not.toBeInTheDocument();
    });
  });
});
