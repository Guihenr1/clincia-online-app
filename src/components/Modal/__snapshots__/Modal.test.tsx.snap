// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders the Modal 1`] = `
<div
  aria-hidden="true"
>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Open modal
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
